From 39937ac422a1ea54deafb8c7d187a61bf695814b Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Tue, 12 Nov 2024 14:15:30 +0000
Subject: [PATCH] dtoverlays: Switch imx462 overlay to use the new compatible

Now that imx462 has a separate compatible string, make use of it.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/imx462-overlay.dts | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

--- a/arch/arm/boot/dts/overlays/imx462-overlay.dts
+++ b/arch/arm/boot/dts/overlays/imx462-overlay.dts
@@ -1,9 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0-only
 // Definitions for IMX462 camera module on VC I2C bus
 
-// IMX462 is the successor to IMX290. The drivers currently don't support
-// any additional feature of IMX462, so use the IMX290 compatible strings
-// for now.
+// IMX462 is the successor to IMX290.
 
 /dts-v1/;
 /plugin/;
@@ -17,19 +15,17 @@
 	// Fragment numbers deliberately high to avoid conflicts with the
 	// included imx290_327 overlay file.
 
-	//IMX462 is not defined in the bindings, so use IMX290 for now.
-
 	fragment@101 {
 		target = <&cam_node>;
 		__overlay__ {
-			compatible = "sony,imx290lqr";
+			compatible = "sony,imx462lqr";
 		};
 	};
 
 	fragment@102 {
 		target = <&cam_node>;
 		__dormant__ {
-			compatible = "sony,imx290llr";
+			compatible = "sony,imx462llr";
 		};
 	};
 
